evm/src/contracts/CircleIntegration/Governance.sol

```diff
@@ -0,0 +1,247 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity ^0.8.22;
+
+import {ERC1967Upgrade} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Upgrade.sol";
+
+import {BytesParsing} from "src/libraries/BytesParsing.sol";
+import {IWormhole} from "src/interfaces/IWormhole.sol";
+import {IGovernance} from "src/interfaces/IGovernance.sol";
+
+import {State} from "./State.sol";
+import {
+    SequentialRegistration,
+    getRegisteredEmitters,
+    getChainToDomain,
+    getConsumedVaas,
+    getDomainToChain,
+    getSequentialRegistration
+} from "./Storage.sol";
+
+abstract contract Governance is IGovernance, State, ERC1967Upgrade {
+    using BytesParsing for bytes;
+
+    error UnsupportedGovernanceAction();
+
+    uint16 constant GOVERNANCE_CHAIN = 1;
+    bytes32 constant GOVERNANCE_EMITTER =
+        0x0000000000000000000000000000000000000000000000000000000000000004;
+
+    // "CircleIntegration" (left-zero-padded)
+    bytes32 constant GOVERNANCE_MODULE =
+        0x000000000000000000000000000000436972636c65496e746567726174696f6e;
+
+    uint8 constant UPDATE_WORMHOLE_FINALITY = 1;
+    uint8 constant REGISTER_EMITTER_AND_DOMAIN = 2;
+    uint8 constant CONTRACT_UPGRADE = 3;
+
+    /**
+     * @notice `updateWormholeFinality` changes the wormhole message consistencyLevel.
+     * @param encodedVaa Attested Wormhole governance message with the following
+     * relevant fields:
+     * - Field Bytes Type Index
+     * - finality 1 uint8 35
+     */
+    function updateWormholeFinality(bytes memory encodedVaa) public pure {
+        // This is here to silence an unused variable warning.
+        encodedVaa;
+
+        revert UnsupportedGovernanceAction();
+    }
+
+    /// @inheritdoc IGovernance
+    function registerEmitterAndDomain(bytes memory encodedVaa) public {
+        (IWormhole.VM memory vaa, uint256 offset) =
+            _verifyAndConsumeGovernanceMessage(encodedVaa, REGISTER_EMITTER_AND_DOMAIN);
+
+        // Registering emitters should only be relevant for this contract's chain ID,
+        // unless the target chain is 0 (which means all chains).
+        uint16 targetChain;
+        (targetChain, offset) = vaa.payload.asUint16Unchecked(offset);
+
+        // NOTE: Reverting with Error(string) comes from the old implementation, so we preserve it.
+        require(targetChain == 0 || targetChain == _chainId, "invalid target chain");
```

Comment:
I think it's ok to remove these old revert strings, since these methods are gated by having a signed governance VAA.

Reply:
Removed the comments referring to reverting with the built-in now that we took out the custom error for the deprecated `updateWormholeFinality` method: https://github.com/wormhole-foundation/wormhole-circle-integration/pull/48/commits/0c80c47c18129d7a679968472a89b96a25e3574e